pkg/syncer/fullsync.go

```diff
@@ -367,11 +407,30 @@ func volumeInfoCRFullSync(ctx context.Context, metadataSyncer *metadataSyncInfor
 		}
 		// Create VolumeInfo CR if not found.
 		if !crExists {
-			err := volumeInfoService.CreateVolumeInfo(ctx, volumeID, vc)
-			if err != nil {
-				log.Errorf("FullSync for VC %s: failed to create VolumeInfo CR for volume %s."+
-					"Error: %+v", vc, volumeID, err)
-				continue
+			if isMultiVCenterFssEnabled && len(metadataSyncer.configInfo.Cfg.VirtualCenter) > 1 {
+				err := volumeInfoService.CreateVolumeInfo(ctx, volumeID, vc)
+				if err != nil {
+					log.Errorf("FullSync for VC %s: failed to create VolumeInfo CR for volume %s."+
+						"Error: %+v", vc, volumeID, err)
+					continue
+				}
+			} else if metadataSyncer.clusterFlavor == cnstypes.CnsClusterFlavorWorkload && isPodVMOnStretchSupervisorFSSEnabled {
+				pv := volumeIdTok8sPVMap[volumeID]
+				pvc, err := metadataSyncer.pvcLister.PersistentVolumeClaims(
+					pv.Spec.ClaimRef.Namespace).Get(pv.Spec.ClaimRef.Name)
+				if err != nil {
+					log.Warnf("Failed to get pvc for namespace %s and name %s. err=%+v",
+						pv.Spec.ClaimRef.Namespace, pv.Spec.ClaimRef.Name, err)
+					continue
+				}
+				pvcCapacity := pvc.Status.Capacity[v1.ResourceStorage]
+				err = volumeInfoService.CreateVolumeInfoWithPolicyInfo(ctx, volumeID, pvc.Namespace,
+					scNameToPolicyIdMap[*pvc.Spec.StorageClassName], *pvc.Spec.StorageClassName, vc, &pvcCapacity)
```

Comment:
Can you add a check for pvc.Spec.StorageClassName != nil?

Reply:
Done